### PR TITLE
feat(datadog): add sampling priority tag in datadog spans

### PR DESCRIPTION
**Pull request by rex-remind101**
encode sampling priority into metrics object of data dog exported data. data dog uses this to sample.

solves https://github.com/open-telemetry/opentelemetry-rust/issues/791

### CODE REVIEW

**Comment by TommyCpp** [comment]
Need to format the code

File: opentelemetry-datadog/src/exporter/model/v05.rs
Comment (TommyCpp) [inline opentelemetry-datadog/src/exporter/model/v05.rs:167]: Could we define some constant at `model/mod.rs` and reuse them across v3 and v5 exporter?

**Review by TommyCpp — CHANGES_REQUESTED** [review]
Have one nit and need to format the code but otherwise LGTM. Thanks for the help!

**Comment by rex-remind101** [comment]
amended my commit with the suggestions. attempted to use the EasyCLA but I'm receiving the following The Individual CLA option is not enabled for this project. Please to your administrator to enable the Individual CLA option for this project. so i'm not sure how to move forward on this.

**Comment by TommyCpp** [comment]
> The Individual CLA option is not enabled for this project. Please to your administrator to enable the Individual CLA option for this project

Could you take a screenshot? It should be something like 
image

**Comment by rex-remind101** [comment]
it worked this time

**Comment by TommyCpp** [comment]
@rex-remind101 I think we need to bump the msrv to 1.56 to resolve the CI issues. Could you make the change in https://github.com/open-telemetry/opentelemetry-rust/blob/main/.github/workflows/ci.yml#L71 ?

I don't have the write access to the branch.

**Comment by rex-remind101** [comment]
@TommyCpp ready!

**Comment by rex-remind101** [comment]
@TommyCpp looks like there is still an error actually, could you help provide direction?

**Comment by TommyCpp** [comment]
> @TommyCpp looks like there is still an error actually, could you help provide direction?

I think the problem is still the MSRV. Let me experiment with some versions and see which one we need to use.

**Review by TommyCpp — APPROVED** [review]
Sorry for the delay. LGTM. Thanks!

**Pull request merged**
